index.mjs

```diff
@@ -69,14 +78,23 @@ export function $(pieces, ...args) {
 }
 
 $.verbose = true
-// Try `command`, should cover all Bourne-like shells.
-// Try `which`, should cover most other cases.
-// Try `type` command, if the rest fails.
-$.shell = `${execSync('command -v bash || which bash || type -p bash')}`.trim()
-$.prefix = 'set -euo pipefail;'
-$.quote = shq
 $.cwd = undefined
 
+$.quote = (s) => s
+$.prefix = ''
+if ($.shell = which.sync('bash', { nothrow: true })) {
+  $.prefix = 'set -euo pipefail;'
+  $.quote = shq
+} else if ($.shell = whichSyncFallback(['pwsh', 'powershell'])) {
+  // Powershell is found on Windows. It has the required support for -c, unlike
+  // cmd.
+  // Prefer 'pwsh', which points to a newer version, for a higher likelihood of
+  // getting "correct" argument passing (PSNativeCommandArgumentPassing).
```

Comment:
Now there is where a lot of POWERSHELL LORE comes into play. For a short version, see https://github.com/PowerShell/PowerShell/issues/13089#issuecomment-835623238.